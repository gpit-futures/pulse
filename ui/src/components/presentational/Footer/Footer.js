import React, {PureComponent} from 'react';

import footerLogo from '../../../assets/images/ripple-foundation-logo-footer.png'

export default class Footer extends PureComponent {
  static defaultProps = {
    isShowSupportedBy: true,
  };

  render() {
    const { copyright, isShowSupportedBy } = this.props;
    return (
      <footer className="footer">
        <div className="container-fluid">
          {copyright ? <p className="footer-text">{ copyright }</p> : null}
          <div className="footer-povered">
            {isShowSupportedBy ? <span className="footer-povered-text">Supported by</span> : null}
            <a href="/" className="footer-logo">
              <img src={footerLogo} alt="Ripple Icon" className="footer-logo-img" />
            </a>
          </div>
        </div>
      </footer>
    )
  }
}
