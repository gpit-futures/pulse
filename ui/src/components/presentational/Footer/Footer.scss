.footer {
  background-color: #EEEEEE;
  color: #757584;
  font-size: 10px;
  line-height: 29px;
  min-width: $min-width;
  padding: 13px 0 14px;
  width: 100%;
  @include respond-from(xs) {
    font-size: 12px;
  }
  @include respond-from(sm) {
    bottom: 0;
    height: 56px;
    position: fixed;
    z-index: 1000;
  }
}

.footer-powered {
  float: right;
  margin-left: 10px;
  vertical-align: middle;
}

.footer-powered-text {
  display: inline-block;
  margin-right: 5px;
}

.footer-logo { display: inline-block; }

.footer-logo-img {
  display: inline-block;
  height: 29px;
}

.footer-text {
  float: left;
  margin: 0;
  margin-right: 10px;
}
