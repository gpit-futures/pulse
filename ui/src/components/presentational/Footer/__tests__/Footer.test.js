import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import Footer from '../Footer';

Enzyme.configure({ adapter: new Adapter() });

const copyright = 'Transforming Usability';

describe('Component <Footer />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(
      <Footer
        copyright={copyright}
      />);

    expect(component.find('.footer-text')).toHaveLength(1);
    expect(component.find('.footer-povered-text')).toHaveLength(1);

    expect(component.find('.footer-text').text()).toEqual(copyright);
    expect(component.find('.footer-povered-text').text()).toEqual('Supported by');

    expect(component).toMatchSnapshot();

    component.setProps({ copyright: undefined, isShowSupportedBy: false });

    expect(component.find('.footer-text')).toHaveLength(0);
    expect(component.find('.footer-povered-text')).toHaveLength(0);

    expect(component).toMatchSnapshot();
  });
});
