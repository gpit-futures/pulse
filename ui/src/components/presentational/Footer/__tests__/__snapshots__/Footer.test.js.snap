// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Footer /> should renders with props correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="container-fluid"
  >
    <p
      className="footer-text"
    >
      Transforming Usability
    </p>
    <div
      className="footer-povered"
    >
      <span
        className="footer-povered-text"
      >
        Supported by
      </span>
      <a
        className="footer-logo"
        href="/"
      >
        <img
          alt="Ripple Icon"
          className="footer-logo-img"
          src="ripple-foundation-logo-footer.png"
        />
      </a>
    </div>
  </div>
</footer>
`;

exports[`Component <Footer /> should renders with props correctly 2`] = `
<footer
  className="footer"
>
  <div
    className="container-fluid"
  >
    <div
      className="footer-povered"
    >
      <a
        className="footer-logo"
        href="/"
      >
        <img
          alt="Ripple Icon"
          className="footer-logo-img"
          src="ripple-foundation-logo-footer.png"
        />
      </a>
    </div>
  </div>
</footer>
`;
